### PR TITLE
NXPhlite-v3 with Leitwert's serial dma

### PR DESCRIPTION
**Pull request by davids5**
This ties together the Now upstreamed DMA work from @laube and @janokle of leitwert on the  current PX4 nuttx

From: https://github.com/PX4/Firmware/pull/8030

This is a WIP for the NXP flight V3 HW based on the K66. It builds but is incomplete as it is missing key Nuttx support in the Kinetis architecture.

The drivers/kinetis/\* is just stubbed out.  There is SPI, or DMA support

Missing Components
- [x]  FPU support in the OS
- [X] DMA (needs to be made consistent with NuttX CS and upstreamed, should also consider streaming)
- [x] SPI - Completed
- [x] CAN
- [x] VBAT - for bootloader app communication.
- [x] Timers PIT, LPTM, and FTM, PDB - for pwm, capture, tones, and hrt.
- [x] RESET up_systemreset(void) noreturn_function;
  
  Incomplete Drivers:
- [x] PWM supports only 1 chan per timer.
- [x] USB The K66 has 2 USB USB0 is FS (Nuttx has driver) USB1 is HS (No NuttX driver!) 
- [x] UART lacking DMA kinetis_serial_dma_poll(void)
         Tested with RTS/CTS HS (RTS out of CPU is per character - Consider GPIO based SW flow control ala stm32)
-[x] SBUS - tested working
- [x] TERMIOS for serial
- [x] RTC - up_rtc_getdatetime - Config issue fixed
- [x] SDHC DMA disabled will not work with it yet

OS/System dependant SW
- [x] Cannot use FPU yet on kinetis
- [x] Termios
- [x] tcgetattr
- [x] tcsetattr
- [x] cfsetspeed

PX4 FMUV2 Modules removed from build due to No External SPI
- [ ] drivers/bma180
- [ ] drivers/bmi160
  
  NOT Portable YET
- [x] MCU version
- [x] /Firmware/src/modules/systemlib/mcu_version.c
- [ ] drivers/pwm_input
- [ ] drivers/test_ppm
- [ ] systemcmds/hardfault_log - needs bsram
- [x] modules/uavcan
- [x] uavcan
- [x] uavcan_stm32_driver - need Kinetis CAN driver

PX4 Drivers Needing porting
- [ ] drivers/kinetis/*
- [ X] drv_hrt.c 
   - [x] HRT 
   - [x] PPM
- [ ] drv_input_capture.c
- [x] drv_io_timer.c
- [ ] drv_led_pwm.cpp
- [x] drv_pwm_servo.c
- [x] drivers/kinetis/adc
- [x] drivers/kinetis/tone_alarm

- [x] 5/31 - Need to add tone control lib. - Done

### CODE REVIEW

**Comment by dannyfpv** [comment]
good flight on pixhawk mini f450.
1- https://logs.px4.io/plot_app?log=859428c0-2686-498e-ab88-53dc15678737
2- https://logs.px4.io/plot_app?log=00682a3b-322e-46e7-aadf-555b92c0f31f
3- https://logs.px4.io/plot_app?log=69ccc09b-7c26-49b3-b78b-2dfb196329f4
4- https://logs.px4.io/plot_app?log=0f30bb5b-890b-4a1f-9439-98cace8e8baa

**Comment by davids5** [comment]
@dannyfpv - would you please test with 5b0ad4e on FMUv2 Pixhawk with 2 MB and Nxphlite

**Comment by dagar** [comment]
image

**Comment by davids5** [comment]
rebased on master

**Comment by davids5** [comment]
@LorenzMeier - 

I was a tad perplexed as to how the minimal changes done to PX4 to bring in Kinetis, could affect the V2 build spilling over by 1028 bytes.   Boaty indicated it was the ROMFS.

There were 3 lines added to the rcS recently but it caused a 1028 spill over. This makes sense 
From reading https://www.kernel.org/doc/Documentation/filesystems/romfs.tx 
```
Another thing to note is that romfs works on file headers and data
aligned to 16 byte boundaries, but most hardware devices and the block
device drivers are unable to cope with smaller than block-sized data.
To overcome this limitation, the whole size of the file system must be
padded to an 1024 byte boundary.
```

@dagar and I spent some time looking at memory usage. 
There is still the 9.5K to get back from mavlink.
There are 41 K of strings for the logger field decoding. ~3 K can be recovered by encoding the types as ordinals not strings in the `orb_*._fields` 

@dagar volunteered to pymavlink PR doen, that will buy us some time, but may take time to come in upstream.

I can not get this branch merged until master get cleaned has some room.

**Comment by davids5** [comment]
rebased dropping 
`DO NOT MERGE TO MASTER: Lower avionics check thresholds` 0de1f02

File: src/modules/uavcanesc/uavcan_driver.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 20158 PX4 Development Team. All rights reserved.
```
Comment (dagar) [inline src/modules/uavcanesc/uavcan_driver.hpp]: copyright

File: src/modules/uavcanesc/uavcan_driver.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @author David Sidrane <david_s5@nscdgg.com>
+ */
+
+#pragma once
+#if defined(UAVCAN_KINETIS_NUTTX)
```
Comment (dagar) [inline src/modules/uavcanesc/uavcan_driver.hpp]: Can't we set the platform define automatically from ${CONFIG_ARCH_CHIP}?

**Pull request merged**